sdl/functions.go

```diff
@@ -300,6 +300,24 @@ func IOFromConstMem(mem []byte) (*IOStream, error) {
 	return stream, nil
 }
 
+// SDL_IOFromDynamicMem - Use this function to create an SDL_IOStream that is backed by dynamically allocated memory.
+// (https://wiki.libsdl.org/SDL3/SDL_IOFromDynamicMem)
+func IOFromDynamicMem() *IOStream {
+	return iIOFromDynamicMem()
+}
+
+// Creates a stream from an array of bytes.
+func IOFromBytes(b []byte) (*IOStream, error) {
```

Comment:
Can we have this function moved to sdl/helpers.go?

Reply:
Actually², I now think it's fine to have this defined in sdl/helpers.go. It's not sure whether we'll add `IOFromMem` later, and even if we do it's fine to have a duplicate I think!

Reply:
Now that I thought about it more, maybe `IOFromBytes` isn't what is needed. What I really need is something like `IOFromFS` to load in assets that are found in the binary from `embed`. I've removed it for now because I think something else could be better here and there is no need to change the API footprint right now. 

Originally, I was trying `IOFromConstMem` using the byte slice returned by `ReadFile` but the pointer seemed to invalidate as soon as I placed it into a map and returned from the function. 

Any ideas on how to handle this efficiently?

Reply:
> Originally, I was trying IOFromConstMem using the byte slice returned by ReadFile but the pointer seemed to invalidate as soon as I placed it into a map and returned from the function.

Ah right, I think I can see why..
The stream returned from `IOFromConstMem` depends on the slice, but no hard reference to this slice is made in the `IOStream` object, so GC might collect it in the caller code.
So, I think `IOFromConstMem` and `IOFromMem` cannot be made safe, it would require the user to ensure the byte slice used to make the stream is kept alive afterwards, but that's not convenient.

So, I think we want your `IOFromBytes` in the end, as a reliable way (in sdl/helpers.go)!

Reply:
I've pushed an update that includes the `IOFromBytes` function in helpers.go. Let me know if there is anything more that should be done for now. Thanks!

Reply:
That's fine by me, thank you for the PR!